### PR TITLE
#4232 display patient age

### LINKED ISSUE #4232: Show the patients age next to the DOB on the patient edit/create form
## Is your feature request related to a problem? Please describe.

Similar to desktop with an age and dob field next to each other:

image

## Describe the solution you'd like

Two fields next to each other. If you edit one, it updates the other.

## Implementation

- This is probably more annoying than it might seem and may be easier to implement by just showing the age rather than having the age be editable

## Describe alternatives you've considered

N/A

## Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #4232 

## Change summary

- Adds a component to manage dob with an age

## Testing

- [ ] When editing a patient, can edit the age or dob 

### Related areas to think about

N/A

### CODE REVIEW

File: src/widgets/modalChildren/PatientEditModal.js [hunk 1]
```diff
@@ -23,51 +23,59 @@ export const PatientEditModalComponent = ({
...
-          isDisabled={surveySchema && surveyForm ? !nameNoteIsValid : isDisabled}
-          textStyle={styles.saveButtonTextStyle}
-          text={generalStrings.save}
-        />
-        <PageButton
-          onPress={onCancel}
-          style={styles.cancelButton}
-          textStyle={styles.cancelButtonTextStyle}
-          text={modalStrings.cancel}
+  canSaveForm,
```
Comment (joshxg) [inline src/widgets/modalChildren/PatientEditModal.js:26]: off road: The patient edit modal wasn't updating the save button when the form was invalid

Comment (mark-prins) [reply src/widgets/modalChildren/PatientEditModal.js:26]: oh yes. I did wonder about that. thanks!

**Review by mark-prins — APPROVED** [review]
wow, a thing of beauty! Nice use of the action/reducer internal state and also adding in the new component for DOB.

File: src/widgets/FormInputs/FormDOBInput.js [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+
+  const onTypeDate = newDate => {
+    const isValid = onValidate(newDate);
+    typedDate(newDate, isValid);
+    onChangeDate(moment(newDate, DATE_FORMAT.DD_MM_YYYY).toDate());
+  };
+
+  const onTypeAge = newAge => {
+    const asNumber = Number(newAge);
+    const isValid = !Number.isNaN(asNumber) && asNumber < 500;
```
[the remark this answers is not in the document]
Comment (joshxg) [reply src/widgets/FormInputs/FormDOBInput.js:134]: Hopefully it's future proof enough!!!!

**Comment by joshxg** [comment]
Also not sure what I'm basing this off.. I think if I base off develop and we just merge develop into 8.2?

**Comment by mark-prins** [comment]
> Also not sure what I'm basing this off.. I think if I base off develop and we just merge develop into 8.2?

yeah, I thought to merge develop into 8.2 after this one is merged in

**Pull request merged**